modules/4337/certora/specs/SignatureLengthCheck.spec

```diff
@@ -0,0 +1,27 @@
+using Account as safeContract;
+
+methods {
+    function checkSignaturesLength(bytes, uint256) external returns(bool) envfree;
+    function combineBytes(bytes, bytes) external returns(bytes) envfree;
+
+    // Safe Contract functions
+    function safeContract.containsContractSignature(bytes, uint256) external returns(bool) envfree;
+    function safeContract.canonicalSignatureHash(bytes, uint256) external returns(bytes32) envfree;
+}
+
+// This rule verifies that if excess data is added to the dynamic part of the signature, then the signature check will fail.
+rule signatureLengthCheckDirectly(bytes signatures, bytes gasGriefingData, uint256 threshold) {
+    require signatures.length > 0;
+    require gasGriefingData.length > 0;
+    assert checkSignaturesLength(signatures, threshold) => !checkSignaturesLength(combineBytes(signatures,gasGriefingData), threshold);
+}
+
+// This rule verifies that if excess data is added to the dynamic part of the signature, though it could pass in the safe contract's
+// `checkSignatures(...)`, it will be caught within the `_checkSignaturesLength(...)` call, as the dynamic length is checked.
+rule canonicalHashBasedLengthCheck(bytes signatures, bytes griefedSignatures, uint256 threshold) {
+    require safeContract.canonicalSignatureHash(signatures, threshold) == safeContract.canonicalSignatureHash(griefedSignatures, threshold);
+    require signatures.length < griefedSignatures.length;
+    require safeContract.containsContractSignature(signatures, threshold);
```

Comment:
This shouldn't be needed. What counter-example was it finding?

In particular, if you append extra bytes to EOA signatures, you would also get a griefing attack that this function should protect from.

Reply:
You are right, we can avoid this check. Updated 👍🏾